cmake/modules/vvencCompilerSupport.cmake

```diff
@@ -84,6 +85,73 @@ function( _emscripten_enable_wasm_simd128 )
   endif()
 endfunction()
 
+function( _set_if_compiler_supports_sve_flag output_var sve_flag )
+  set_if_compiler_supports_flag( tmp_var "${sve_flag}" )
+  if( NOT tmp_var )
+    return()
+  endif()
+
+  set( OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS} )
+  set( OLD_CMAKE_TRY_COMPILE_TARGET_TYPE ${CMAKE_TRY_COMPILE_TARGET_TYPE} )
+  set( CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} ${sve_flag}" )
+  set( CMAKE_TRY_COMPILE_TARGET_TYPE STATIC_LIBRARY )
+
+  # Check whether the compiler can compile SVE functions that require
+  # backup/restore of SVE registers according to AAPCS.
+  # https://github.com/llvm/llvm-project/issues/80009.
+  set( SVE_COMPILATION_TEST "
+#include <arm_sve.h>
+void other();
+svfloat32_t func(svfloat32_t a) {
+other();
+return a;
+}
+int main() { return 0; }" )
+
+  check_c_source_compiles( "${SVE_COMPILATION_TEST}" SVE_COMPILATION_C_TEST_COMPILED )
+  check_cxx_source_compiles( "${SVE_COMPILATION_TEST}" SVE_COMPILATION_CXX_TEST_COMPILED )
+
+  # Check if arm_neon_sve_bridge.h is available.
+  set( SVE_HEADER_TEST "
+#ifndef __ARM_NEON_SVE_BRIDGE
+#error 1
+#endif
+#include <arm_sve.h>
+#include <arm_neon_sve_bridge.h>
+int main() { return 0; }")
+  check_c_source_compiles( "${SVE_HEADER_TEST}" SVE_HEADER_C_TEST_COMPILED )
+  check_cxx_source_compiles( "${SVE_HEADER_TEST}" SVE_HEADER_CXX_TEST_COMPILED )
+
+  set( CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS} )
+  set( CMAKE_TRY_COMPILE_TARGET_TYPE ${OLD_CMAKE_TRY_COMPILE_TARGET_TYPE} )
+
+  if( SVE_COMPILATION_C_TEST_COMPILED AND SVE_COMPILATION_CXX_TEST_COMPILED AND
+      SVE_HEADER_C_TEST_COMPILED AND SVE_HEADER_CXX_TEST_COMPILED )
+    set( ${output_var} TRUE PARENT_SCOPE )
+  endif()
+endfunction()
+
+# Check if the compiler supports the AArch64 SVE and SVE2 extensions, and set
+# variables for flags used to enable them to avoid duplication.
+function( set_if_compiler_supports_arm_extensions output_var_sve flag_sve output_var_sve2 flag_sve2 )
+  if( NOT(( ${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64" ) OR
+          ( ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" )))
+    return()
+  endif()
+  if( UNIX OR MINGW )
+    # SVE is an optional feature from Armv8.2-A.
+    set( ${flag_sve} "-march=armv8.2-a+sve" )
+    set( ${flag_sve} "-march=armv8.2-a+sve" PARENT_SCOPE )
+    _set_if_compiler_supports_sve_flag( ${output_var_sve} "${${flag_sve}}" )
+    set( ${output_var_sve} "${${output_var_sve}}" PARENT_SCOPE )
```

Comment:
This looks overly complicated. Do I understand correctly? You create a *local* variable with the name provided as a function parameter to `set_if_compiler_supports_arm_extensions()` and store the variable *name* in `flag_sve`. You then pass the name of that variable to `_set_if_compiler_supports_sve_flag()` to be overwritten? I think, if you introduced a local variable with a different name, all these complex indirections can be avoided.

Wouldn't something like this also work? Untested:
```cmake
    set( _flag_sve "-march=armv8.2-a+sve" )
    _set_if_compiler_supports_sve_flag( _sve_supported "${_flag_sve}" )
    if( _sve_supported )
      set( ${flag_sve} "${_flag_sve}" PARENT_SCOPE )
      set( ${output_var_sve} "${_sve_supported}" PARENT_SCOPE )
    endif()
```

additionally, if I am not mistaken, the variable in `flag_sve` will always contain `-march=armv8.2-a+sve` regardless of whether it is supported or not.

My function `set_if_compiler_supports_flag()` is intended to work like this:
You pass an output variable and a flag, and if the compiler does support the flag, the output variable will contain the flag, so you can just use the variable in the compile options. I would prefer if your similarly named function behaved somehow similar.
It's fine for me if you like to add a separate boolean output variable to explicitly convey whether SVE is supported, but at least in respect to the flag variable it should behave similar.

Reply:
Thanks for the suggestions, that does simplify things quite a bit.